openedx_proversity_reports/utils.py

```diff
@@ -13,49 +13,46 @@
 from openedx_proversity_reports.edxapp_wrapper.get_supported_fields import get_supported_fields
 
 
-def generate_report_as_list(users, course_key, block_report_filter):
+def generate_report_as_list(users, course_key, block_report_filter, root_block):
     """
     Returns a list with the user information for every block in block_report_filter.
     """
 
-    def verify_name(name, data):
+    def update_user_dict(child, user_data, section=None, subsection=None, vertical=None):
         """
-        This verifies if the name is already in use and generates a new one.
+        Returns the complete data for the given values.
         """
-        if name in data:
-            identifier = name.split('-')[-1]
-            try:
-                number = int(identifier) + 1
-                name = name.replace(identifier, str(number))
-            except ValueError:
-                name = u"{}-{}".format(name, "1")
+        if child.get('type') in block_report_filter:
 
-            name = verify_name(name, data)
+            aux = user_data.get(child.get('type'), [])
 
-        return name
+            child_data = dict(
+                name=child.get('display_name'),
+                complete=child.get('complete'),
+                number=child.get('position_number'),
+            )
 
-    def update_user_dict(child, user_data, parent=None):
-        """
-        Returns the complete data for the given values.
-        """
-        if child.get("type") in block_report_filter:
+            if section and section != child:
```

Comment:
It is not necessary to check section, because if section is None it's different to child. @andrey-canon 
Same comment for subsection and vertical.

Reply:
Addressed.